### DISCUSSION TITLE
Problem switching from Liquid to Nunjucks

### DISCUSSION THREAD

**Discussion by arpeasal** [11ty/eleventy#2330] [Q&A] [answered]
I just tried switching my 11ty project from Liquid to Nunjucks.  I changed all the file extentions (both in the actual file names and references to those files inside of files).  When I did this, I typed the first change and copied it and then pasted it everywhere else, so it's doubtful I made a typing error.  Everywhere there was the {{ content }} variable, I changed it to {{ content | safe }}.  The site seemed to work fine, I didn't find any errors, UNTIL ...

I tried using a "set tag" and got an error: `[11ty] > Having trouble rendering liquid template ./source/index.md`

I tried an experiment.  I added the two lines of code below to a previous project that was done in nunjucks.
```
{% set some_var1 = "Foo" %}
{{ some_var1 }}
```
As expected, this printed out Foo where expected.

I then tried the same thing in my current project and again got the "Having trouble rendering liquid template" error.

I can't figure out why it thinks I'm using a liquid template.  I can't find any reference to liquid in the project, including in the .eleventy.js  file.  What else do I need to do to switch from liquid to nunjucks?

**Comment by cfjedimaster** [comment, marked as answer]
Markdown is processed by Liquid by default (https://www.11ty.dev/docs/languages/markdown/). Did you change your configuration to process *.md as Nunjucks?

**Reply by pdehaan** [reply]
Also https://www.11ty.dev/docs/config/#default-template-engine-for-markdown-files

**Reply by arpeasal** [reply]
Thanks to both of you.
I had looked at the .eleventy.js file from the project I mentioned that the set tag worked in and didn't see anything that looked like it was configuring nunjucks.
After reading your replies and the info at the pages you referenced, I added 2 templating lines to my .eleventy.js file.  The bottom of how that file looks now is below.  I'm not sure if I needed to add both the markdown AND HTML templating lines, but I think it is ok.

The project still isn't working, but at least I'm now getting errors that say "njk" instead of "liquid".  I'll work through the errors and hopefully I'll get it working.  So far the error seem to be mostly with filters.
 
 ```
 return {
    markdownTemplateEngine: "njk",
    htmlTemplateEngine: "njk",
    dir: {
      input: "source",
      output: "public",
    },
  };
};
```
